storages/file-storage/src/store.rs

```diff
@@ -0,0 +1,95 @@
+use {
+    crate::{FileRow, FileStorage, ResultExt},
+    async_trait::async_trait,
+    futures::stream::iter,
+    gluesql_core::{
+        data::{Key, Schema},
+        error::Result,
+        store::{DataRow, RowIter, Store},
+    },
+    std::fs,
+};
+
+#[async_trait(?Send)]
+impl Store for FileStorage {
+    async fn fetch_all_schemas(&self) -> Result<Vec<Schema>> {
+        let mut schemas = fs::read_dir(&self.path)
+            .map_storage_err()?
+            .map(|dir_entry| {
+                let dir_entry = dir_entry.map_storage_err()?;
+                let file_type = dir_entry.file_type().map_storage_err()?;
+                if file_type.is_dir() {
```

Comment:
```suggestion
                let path = dir_entry.path();
                let extension = path.extension().and_then(OsStr::to_str);
                if file_type.is_dir() || extension != Some("sql") {
```

Would it be better to validate if it's *.sql as well?

Reply:
Or.. maybe we can reuse fetch_schema